### ISSUE TITLE
Add openssl to the base system

### ISSUE THREAD

**Issue by allaeddineomc** [fedora-silverblue/issue-tracker#201]
a lo of programs require open SSL to function properly and it's not included by default  , i wish you add it so i can use GSconnect shell extension without overlaying packages

**Comment by juhp** [comment]
Do you mean `/usr/bin/openssl`?

**Comment by allaeddineomc** [comment]
yes

**Comment by felipeborges** [comment]
https://github.com/GSConnect/gnome-shell-extension-gsconnect/wiki/Help#openssl-not-found

**Comment by travier** [comment]
Overlaying packages is fine, but this could be a good enough reason to add openssl.

**Comment by allaeddineomc** [comment]
i know it's fine , but as the best just works distro ever existed it's a shame to make users use the terminal to have such basic functionality , i think gsconnect is a must in all gnome distros but i won't complain about not including it because installing it isn't confusing for noobs

**Comment by HarryMichal** [comment]
Opened https://pagure.io/workstation-ostree-config/pull-request/260

**Comment by travier** [comment]
See https://pagure.io/workstation-ostree-config/pull-request/260#comment-176049. We need someone to start a discussion with the Workstation Working Group (https://pagure.io/fedora-workstation/issues) to move forward.

**Comment by osalbahr** [comment]
Any updates? I'm fine overlaying `openssl `in fedora silverblue, but I'd prefer to avoid that. Any reasons why it shouldn't be preinstalled in fedora silverblue or fedora workstation?

```console
@fedora ~]$ open ssl
bash: open: command not found
@fedora ~]$ uname -a
Linux fedora 6.2.9-300.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Thu Mar 30 22:32:58 UTC 2023 x86_64 GNU/Linux
@fedora ~]$ 
```

**Comment by juhp** [comment]
How about packaging the extension in Fedora, that may be better UX?
I suspect the majority of Fedora users don't need/use /usr/bin/openssl.

**Comment by tpopela** [comment]
> Any updates? 

No updates as no-one didn't open the request as Timothee was suggesting - see that https://pagure.io/fedora-workstation/issues?status=Open&search_pattern=openssl&close_status= returns 0 issues.

**Comment by strugee** [comment]
> How about packaging the extension in Fedora, that may be better UX?

I'm not sure this is true - the original impetus for this issue was to be able to use GSConnect without overlaying packages. I mean, _maybe_ this suggestion would be better since at least it would show up as a graphical app in GNOME Software...? But that still isn't a great experience for people who expect the extensions.gnome.org version to Just Work™.

**Comment by Namoria** [comment]
Your average Fedora Silverblue user reporting in; I don’t miss any packages in Silverblue and have none layered because I am perfectly fine with using Flatpaks and a simple toolbox – _except_ OpenSSL.
GSConnect is a must-have GNOME extension on my desktop and on my parents’ notebooks. I’d love to see it included in the image in the future. That would be awesome.

**Comment by TheOneandOnlyJakfrost** [comment]
Since there is still no issue for this raised with the Workstaion working group, should I open one and link it to this issue?

**Comment by travier** [comment]
Yes please. See: https://discussion.fedoraproject.org/t/why-not-openssl-bundled-with-silverblue/112726/5

**Comment by TheOneandOnlyJakfrost** [comment]
Created issue to be found at https://pagure.io/fedora-workstation/issue/426

**Comment by travier** [comment]
From https://pagure.io/fedora-workstation/issue/426#comment-906574:

```
We have a very long discussion about this today. We have multiple recommendations:

Install openssl yourself
Ask Silverblue to preinstall openssl
Enhance GSConnect to use certtool (the GnuTLS certificate generation tool), which is already installed by default, if available
Installing the openssl command line tool would be harmless and not objectionable, but we don't want to preinstall every potentially useful utility. openssl command line is redundant with certtool.
```

**Comment by TheOneandOnlyJakfrost** [comment]
Well, I think the GSConnect route is a good way to go, at least since certtool is already a part of WS.

**Comment by strugee** [comment]
@travier is this a good first issue? By my reading there isn't a clearly defined path forward

**Comment by travier** [comment]
The next step would be to reach out to GSConnect to see if they could add certtool as an alternative to openssl in the GNOME extension. If someone wants to try adding this support that would be great.

There are only one(?) call to openssl in the code as far I can find so this should be a relatively small change: https://github.com/search?q=repo%3AGSConnect%2Fgnome-shell-extension-gsconnect%20openssl&type=code

**Comment by chrisawi** [comment]
I just noticed that `certtool` isn't in the Silverblue base image either (`gnutls-utils`). On Workstation (at least in my old F38 VM), it looks like it's pulled in by the libvirt stack for gnome-boxes.

**Comment by travier** [comment]
I've created a sysext for openssl: https://github.com/travier/fedora-sysexts/tree/main/openssl

That should make it easier to temporarily get openssl to setup the connection with GSConnect and then remove it, instead of layering / removing.
